bin/opteadm/src/bin/opteadm.rs

```diff
@@ -244,14 +244,19 @@ enum Command {
         external_net: ExternalNetConfig,
     },
 
-    /// Allows a guest to send and receive traffic on a given CIDR block.
+    /// Allows a guest to send or receive traffic on a given CIDR block.
     AllowCidr {
         /// The OPTE port to configure
         #[arg(short)]
         port: String,
 
         /// The IP block to allow through the gateway.
         prefix: IpCidr,
+
+        /// Control whether traffic on the CIDR should be allowed for
+        /// inbound or outbound traffic.
+        #[arg(long = "dir")]
+        direction: Direction,
```

Comment:
Given `opteadm` is primarily a dev tool, I think making this an `Option<Direction>` and doing the ioctl calls for both directions if it's `None` is fine.